File: 10_laba/src/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/model/logic/JsonToArrayObject.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic;

import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.AirTicketOfficeContext;
import com.google.gson.Gson;
import com.google.gson.stream.JsonReader;

import java.io.*;

public class JsonToArrayObject {

    private final static String PATH_TO_FILE = "D:\\java\\10_laba\\file.json";

    public static AirTicketOfficeContext DeserializeJson() throws FileNotFoundException {
        JsonReader reader = new JsonReader(new FileReader(PATH_TO_FILE));
        Gson gson = new Gson();
        return gson.fromJson(reader, AirTicketOfficeContext.class);
    }

    public static void SerializeJson(AirTicketOfficeContext contex) throws IOException {
        Gson gson = new Gson();
        FileWriter writer =new FileWriter(PATH_TO_FILE);
        String string = gson.toJson(contex);
        writer.write(string);
        writer.close();
    }
}

File: 10_laba/C##/AirTicketOffice/AirTicketOffice/by/bntu/fitr/java/ingoslingsfootsteps/model/logic/JsonToArrayObject.cs
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.logic
{
    class JsonToArrayObject
    {
        private const string PATH_TO_FILE = "D:\\java\\10_laba\\C##\\file.json";

        public static AirTicketOfficeContext DeserializeJson()
        {
            using (StreamReader file = new StreamReader(PATH_TO_FILE))
            {
                var json = file.ReadToEnd();
                return JsonConvert.DeserializeObject<AirTicketOfficeContext>(json);
            }
        }

        public static void SerializeeJson(AirTicketOfficeContext contex)
        {
            using (StreamWriter file = new StreamWriter(PATH_TO_FILE))
            {
                var json = JsonConvert.SerializeObject(contex);
                file.WriteLine(json);
            }
        }
    }
}
